### ISSUE TITLE
Extremely bad performance on Cityscapes

### ISSUE THREAD

**Issue by ShengkaiWu** [hustvl/SparseInst#38]
Hi, have you ever carry  out experiments on cityscapes datasets? The performance of SparseInst on cityscapes is very bad.
![image](https://user-images.githubusercontent.com/26599885/173507769-8f53a8c2-7631-44f0-a0be-e159443d1fca.png)
All the training settings are adopt as the same as that in COCO. And I have modified the original code to solve the bug when there is no instances in the image. Do you have any idea about the phenomenon?

**Comment by wondervictor** [comment]
Hi @ShengkaiWu, thanks for your interest in SparseInst!
We've already evaluated the SparseInst on Cityscapes (Instance Segmentation) and obtained some tentative results.
Cityscapes results of `SparseInst-GIAM w/ R-50-vd-DCN`:
|AP[val]|AP|AP50|
|---|---|---|
|36.1|31.5|55.1|

For Cityscapes, we adjust the training schedule by: `max_iter=60k, batch_size=8, lr=6e-6`. The standard schedule for Mask R-CNN is not appropriate for SparseInst.
The schedule can be as a reference and we'll continue to improve the performance of SparseInst on both COCO and Cityscapes.

**Comment by ShengkaiWu** [comment]
For training SparseInst on cityscapes, I adopt the model weights trained on COCO to initialize the model. Now, APval on cityscapes is 21% for SparseInst_r50_base. All the settings are adopt the same as you train SparseInst on COCO. Can you share the config file?

**Comment by wondervictor** [comment]
Hi @ShengkaiWu, we can provide you with a config for reference.

The reference config is the below:

```python
_BASE_: "Base-SparseInst.yaml"
MODEL:
  WEIGHTS: ""
  BACKBONE:
    NAME: "build_resnet_vd_backbone"
    FREEZE_AT: 0
  RESNETS:
    NORM: "FrozenBN"
    DEPTH: 50
    STRIDE_IN_1X1: False
    DEFORM_ON_PER_STAGE: [False, False, True, True] # on Res4,Res5
  SPARSE_INST:\
    DECODER:
      NUM_CLASSES: 8
      NAME: "GroupIAMDecoder"
DATASETS:
  TRAIN: ("cityscapes_fine_instance_seg_train",)
  TEST:  ("cityscapes_fine_instance_seg_val",)
SOLVER:
  IMS_PER_BATCH: 8
  BASE_LR: 0.000006
  STEPS: (60000, )
  MAX_ITER: 72000
  WEIGHT_DECAY: 0.001
TEST:
  EVAL_PERIOD: 9000
INPUT:
  MIN_SIZE_TRAIN: (800, 832, 864, 896, 928, 960, 992, 1024)
  MIN_SIZE_TRAIN_SAMPLING: "choice"
  MIN_SIZE_TEST: 1024
  MAX_SIZE_TRAIN: 2048
  MAX_SIZE_TEST: 2048
  FORMAT: "RGB"
OUTPUT_DIR: 'output/sparse_inst_res50vd_dcn_mask_iou_coord_ce_maskbce_unified_light_k128_bcex5_cityscapes_fix_decay'

```

In addition, you might need to load the COCO pre-trained weights. We'd like to release the configs and models for Cityscapes later.

**Comment by cici-999** [comment]
Hello, did you convert cityscapes dataset into coco format? Could you please provide the transformed data set? Thank you.
